snap7/util.py

```diff
@@ -642,25 +644,35 @@ def set_time(bytearray_: bytearray, byte_index: int, time_string: str) -> bytear
 
         Examples:
             >>> data = bytearray(4)
-            >>> snap7.util.set_dint(data, 0, '-22:3:57:28.192')
+
+            >>> snap7.util.set_time(data, 0, '-22:3:57:28.192')
+
             >>> data
                 bytearray(b'\x8d\xda\xaf\x00')
         """
-    import re
     sign = 1
     bits = 32
     data_list = re.split('[: .]', time_string)
-    days, hours, minutes, seconds, milli_seconds = [int(x) for x in data_list]
-    if days < 0:
+    days: str = data_list[0]
+    hours: int = int(data_list[1])
+    minutes: int = int(data_list[2])
+    seconds: int = int(data_list[3])
+    milli_seconds: int = int(data_list[4])
+    if re.split(r'(\d+)$', days)[0:2][0] == '-':
```

Comment:
The same, but a little more readable, imho.
```suggestion
    if re.match('^-\d{1,2}$', days):
```